### ISSUE TITLE
Allow Kibana endpoint configuration to accept custom headers

### ISSUE THREAD

**Issue by igor-kupczynski** [elastic/apm-server#4065]
Add `headers` config map option to to [Kibana endpoint configuration](https://www.elastic.co/guide/en/apm/server/master/setup-kibana-endpoint.html#_kibana_endpoint_configuration_options).

The configuration can look similar to [**headers** option in _Elasticsearch output_](https://www.elastic.co/guide/en/apm/server/master/setup-kibana-endpoint.html#_kibana_endpoint_configuration_options), e.g.

```yaml
apm-server.kibana.headers:
  X-My-Header: "Header contents"
```

The custom HTTP headers should be add to each request directed to the Kibana endpoint.

To align it with the _Elasticsearch output_ [note](https://www.elastic.co/guide/en/apm/server/master/setup-kibana-endpoint.html#_kibana_endpoint_configuration_options):
> It is possible to specify multiple header values for the same header name by separating them with a comma.

This option can be useful in cases where you need to annotate your http traffic, e.g. custom security solutions, tracing proxies, etc.

**Comment by shubhaat** [comment]
@axw would you all be inclined to backport this to 7.9 as well? I ask because we are trying to coordinate how to prioritize the dependencies on our end compared to other issues. If the stack team chooses to merge the changes for this in 7.10 (Which is scheduled for Nov 3, we will schedule this accordingly, but if this might be backported, we would want to schedule something soon.

**Comment by axw** [comment]
@shubhaat the server team discussed this issue yesterday, and we think it should be backported for 7.9.1. I'll check with the Beats team (the code to be changed is in libbeat), but I think this is significant enough to warrant backporting. I'll come back to you.

**Comment by axw** [comment]
@shubhaat confirming we will backport to 7.9.1. There's a fix pending review at https://github.com/elastic/beats/pull/20580.

**Comment by shubhaat** [comment]
@igor-kupczynski and @franekrichardson - this gives us reason to coordinate the fixes for this on Cloud.

**Comment by axw** [comment]
Not done yet, need to update the beats module.

**Comment by axw** [comment]
Changes are in master now, backport to 7.x is https://github.com/elastic/apm-server/pull/4088, and will backport to 7.9 once the libbeat change is merged into the 7.9 branch.

**Comment by igor-kupczynski** [comment]
Brilliant, thanks @axw ! We're working on the cloud part of it; hope we'll be ready some time next week (but as usual, no promises)
